math/mathcore/test/testVectorizedTMath.cxx

```diff
@@ -0,0 +1,78 @@
+#include <random>
+#include "VectorizedTMath.h"
+#include <gtest/gtest.h>
+
+#if defined(R__HAS_VC)
+
+Double_t uniform_random(Double_t a, Double_t b)
+{
+  return a + (b - a) * drand48();
+}
+
+class VectorizedTMathTest : public ::testing::Test
+{
+protected:
+  VectorizedTMathTest() {}
+
+  size_t kVS = vecCore::VectorSize<ROOT::Double_v>();
+  Double_t input_array[16384] __attribute__((aligned(VECCORE_SIMD_ALIGN)));
+  Double_t output_array[16384] __attribute__((aligned(VECCORE_SIMD_ALIGN)));
+};
+
+#define TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, a, b, tolerance)  \
+  TEST_F(VectorizedTMathTest, tmathfunc)                       \
+    {                                                          \
+      int trials = 16384;                                      \
+      for(int i = 0;i < trials;i++)                            \
+        input_array[i] = uniform_random(a, b);                 \
+                                                               \
+      ROOT::Double_v x, y;                                     \
+      for(int j = 0; j < trials;j+=kVS) {                   \
+        vecCore::Load<ROOT::Double_v>(x, &input_array[j]);     \
+        y = TMath::tmathfunc(x);                               \
+        vecCore::Store<ROOT::Double_v>(y, &output_array[j]);   \
+      }                                                        \
+      for(int j = 0; j < trials;j++) {                                       \
+        EXPECT_NEAR(output_array[j], TMath::tmathfunc(input_array[j]), tolerance);  \
+      }                                                                         \
+    }
+
+#define TEST_VECTORIZED_TMATH_FUNCTION_FLT_RANGE(tmathfunc, tolerance)    \
+   TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, -FLT_MAX/2, FLT_MAX/2, tolerance)
+
+#define TEST_VECTORIZED_TMATH_FUNCTION_FLT_POS_RANGE(tmathfunc, tolerance)    \
+   TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, FLT_MIN, FLT_MAX, tolerance)
+
+#define TEST_VECTORIZED_TMATH_FUNCTION_FLT_LOG(tmathfunc, tolerance)      \
+   TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, 1, FLT_MAX, tolerance)
+
+#define TEST_VECTORIZED_TMATH_FUNCTION_FLT_EXP(tmathfunc, tolerance)    \
+   TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, FLT_MIN_10_EXP, FLT_MAX_10_EXP, tolerance)
+
+#define TEST_VECTORIZED_TMATH_FUNCTION_FLT_EXP_POS(tmathfunc, tolerance)    \
+   TEST_VECTORIZED_TMATH_FUNCTION(tmathfunc, FLT_MIN, FLT_MAX_10_EXP, tolerance)
+
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_LOG(Log2, 1e-13);
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_RANGE(BreitWigner, 1e-15);
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_RANGE(Gaus, 1e-15);
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_EXP(LaplaceDist, 1e-15);
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_EXP(LaplaceDistI, 1e-15);
+TEST_VECTORIZED_TMATH_FUNCTION(Freq, -6, 6, 1e-15);
+TEST_VECTORIZED_TMATH_FUNCTION_FLT_EXP(KolmogorovProb, 1.3e-10);
```

Comment:
Regarding `KolmogorovProb` the test passes with tolerance of `1.3e-10`. With lower tolerance the issues arises in the case where `u < 0.755`. As far as I can see, the vectorized translation of that case seems right.